pkg/validate/validate.go

```diff
@@ -107,25 +107,63 @@ func ImagePullSecret(secret string) error {
 // the ClusterName function to confirm compliance with the rest. The resulting name will
 // therefore match RFC1035 with the exception of possible periods ".", which will be
 // translated into dashes "-" in the InfraID before being used to create cloud resources.
-func ClusterName1035(v string) error {
-	re := regexp.MustCompile("^[a-z]")
-	if !re.MatchString(v) {
-		return errors.New("cluster name must begin with a lower-case letter")
+// The optional param m is the max length for the string which the platform support.
+func ClusterName1035(v string, m ...int) error {
```

Comment:
Do not use a variadic function as a way of overloading the function name. Instead, add a new function `ClusterName1035WithMaxLength` and have the existing `ClusterName1035` call the new function.

```
func ClusterName1035(v string) error {
  const maxlen = validation.DNS1123LabelMaxLength - 9
  ClusterName1035WithMaxLength(v, maxlen)
}
```